code/modules/mob/living/living_defense.dm

```diff
@@ -89,15 +89,13 @@
 
 	//Armor
 	var/absorb = run_armor_check(def_zone, P.check_armour, P.armor_penetration)
-	var/proj_sharp = is_sharp(P)
-	var/proj_edge = has_edge(P)
 	var/damaged
-	if ((proj_sharp || proj_edge) && prob(absorb))
-		proj_sharp = 0
-		proj_edge = 0
+	if ((P.damage_flags & DAM_SHARP || P.damage_flags & DAM_SHARP) && prob(absorb))
+		P.damage_flags &= ~DAM_SHARP
+		P.damage_flags &= ~DAM_EDGE
```

Comment:
Why is this modifying an object's damage flags? Isn't there a chance they'll be used elsewhere without it imploding?

Reply:
That's just for projectiles.